### DISCUSSION TITLE
How to set step size when dragging a DragValue or Slider?

### DISCUSSION THREAD

**Discussion by 8bittree** [emilk/egui#639] [Q&A]
I've got a value that needs to be a multiple of 256 and within a certain range so DragValue or Slider seem like they would be a good widget for it. However, I'm not seeing a straightforward way of getting the step size of either of them to 256.

For DragValue, I tried setting the speed to 256, but that for some strange reason made the step size 500 (or 50 with shift), resulting in a cornucopia of invalid values. Without setting a speed, it is possible to hit multiples of 256, but there's an annoying 255 invalid values to jump through in between.

Slider doesn't seem to have any option to directly set a step size and by default is stuck to multiples of 100 (or 50 with shift). Turning off smart aim at least makes it possible to hit multiples of 256, but then it has the same annoyance as the default DragValue.

Is there something I'm missing, or is this a strict (intended?) limitation of those widgets?

**Comment by Gordon01** [comment]
I need this feature too and made a PR with `step_by()` method added to Slider: https://github.com/emilk/egui/pull/1225
